docs/src/main/asciidoc/native-reference.adoc

```diff
@@ -153,6 +153,299 @@ $ quarkus-project-0.1-SNAPSHOT-runner -XX:PrintFlags=
+  the very first allocation that 36 threads do is taking the thread name and trimming it so that it can fall within the 15 character limit enforced by kernels.
+  For each of those allocations,
+  there are 2 `mmap` calls,
+  one to reserve the memory and the other to commit it.
+* If you capture the `syscalls:sys_enter_munmmap` event,
+  you might observe that some allocations also result in calls to `munmap`.
+  When calculating the size to reserve,
+  the requested size to allocate can be rounded up to the page size.
+  The maintain alignment,
+  1MB in case of aligned chunks or 1 byte for unaligned chunks,
+  some of the reserved memory might be unreserved.
+  That is where these `munmap` calls would come from.
+
+[TIP]
+====
+Just by looking at the flamegraph and counting the number of `mmap` calls that originate in thread local allocations,
+an approximate estimation can be made on the RSS a Quarkus native executable uses on startup.
+This estimation can't be fully precise because for unaligned heap chunks, the stack trace doesn't record the size of the chunk.
+Also, if the additional build args are removed,
+inlining optimizations might obscure the type of array allocation chunk requested,
+so it might not be possible to decipher if the allocation is for an aligned or unaligned chunk.
+====
+
+[[rss-ebpf]]
+==== bcc/eBPF
+
+A version of
+https://github.com/iovisor/bcc[bcc]/
+https://ebpf.io/[eBPF]
+that can do stack traces is only available from Linux kernel 4.8 onwards.
+It can do in-kernel summaries,
+which makes it more efficient and has lower overhead.
+Unfortunately it doesn't understand DWARF debug symbols,
+so the information obtained might be harder to read and contain less detail.
+
+bcc/eBPF is very extensible so it's easier to tailor make scripts to track specific metrics.
+The `bcc` project contains a `stackcount` program that can be used to count stack traces in similar way to what `perf` did in the previous section.
+But in some cases, it might be more useful to have other metrics other than number of calls to a system call.
+`malloc` is one such example.
+The number of `malloc` calls is not so important,
+but rather the size of the allocations.
+So rather than having a flamegraph showing sample counts,
+a flamegraph can be generated that shows bytes allocated.
+
+Aside from `mmap`,
+`malloc` system calls are also present in GraalVM native executables.
+Let's put bcc/eBPF in action to generate a flamegraph of bytes allocated using `malloc`.
+
+To do this,
+first re-generate a Quarkus native executable removing debug info,
+which bcc/eBPF does not understand,
+and instead use frame pointer with local symbols to get the stack traces:
+
+[source,bash]
+----
+$ mvn package -DskipTests -Dnative \
+  -Dquarkus.native.additional-build-args=-H:-DeleteLocalSymbols,-H:+PreserveFramePointer
+----
+
+https://github.com/galderz/BPF-tools/blob/topic.1202.fix-mallocstacks/old/2017-12-23/mallocstacks.py[mallocstacks.py]
```

Comment:
Please upload the script in the guide's folder structure to avoid external links.

Reply:
Good idea. I don't want to embed it necessarily but definitely good idea having it there. I had to slightly tweak the original script from Brendan to work with more recent Python versions.

Reply:
Added script and pushed update.